src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -58,6 +60,70 @@ std::string MutationReplaceBinaryOperator::GetExpr(
   return result;
 }
 
+bool MutationReplaceBinaryOperator::IsRedundantReplacementOperator(
+    clang::BinaryOperatorKind op, clang::ASTContext& ast_context) const {
+  clang::Expr::EvalResult rhs_eval_result;
+  clang::Expr::EvalResult lhs_eval_result;
+
+  bool rhs_is_int =
+      binary_operator_.getRHS()->EvaluateAsInt(rhs_eval_result, ast_context);
+  bool lhs_is_int =
+      binary_operator_.getLHS()->EvaluateAsInt(lhs_eval_result, ast_context);
+
+  // In the case where both operands are 0, the only case that isn't covered
+  // by constant replacement is undefined behaviour, this is achieved by /.
+  if (rhs_is_int &&
+      llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
+                                llvm::APSInt::get(0)) &&
+      lhs_is_int &&
+      llvm::APSInt::isSameValue(lhs_eval_result.Val.getInt(),
+                                llvm::APSInt::get(0))) {
+    if (op == clang::BO_Div) {
+      return false;
+    }
+  }
+
+  // In the following cases, the replacement is equivalent to either replacement
+  // with a constant or argument replacement.
+  if (rhs_is_int && llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
+                                              llvm::APSInt::get(0))) {
+    // When the right operand is 0: +, -, << and >> are all equivalent to
+    // replacement with the right operand; * is equivalent to replacement with
+    // the constant 0 and % is equivalent to replacement with /.
+    if (op == clang::BO_Add || op == clang::BO_Sub || op == clang::BO_Shl ||
+        op == clang::BO_Shr || op == clang::BO_Mul || op == clang::BO_Rem) {
+      return true;
+    }
+  } else if (rhs_is_int &&
+             llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
+                                       llvm::APSInt::get(1))) {
+    // When the right operand is 1: * and / are equivalent to replacement by the
+    // left operand.
+    if (op == clang::BO_Mul || op == clang::BO_Div) {
+      return true;
+    }
+  }
+
+  if (lhs_is_int && llvm::APSInt::isSameValue(lhs_eval_result.Val.getInt(),
```

Comment:
Same comment as my nit-picky one above is relevant here.

Reply:
Same reply as above.